vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,225 @@
+     * @param exhibitionFolderName - the name of the folder where the exhibition data is stored
+     * @param sequenceHistory - the history of sequences to be included in the snapshot
+     * @param exhibitionSnapshot - the snapshot object that will store the exhibition state
+     * @return 
+     * @throws IOException - if an I/O error occurs during the snapshot creation
+     * @throws InterruptedException - if the snapshot creation process is interrupted
+     * @throws FileStorageException - if an error occurs while storing the snapshot
+     * @throws ImageCouldNotBeStoredException 
+     */   
+    @Override
+    @Async
+    @Transactional
+    public SnapshotTask createSnapshot(String resourcesPath, String exhibitionFolderName,SequenceHistory sequenceHistory, ExhibitionSnapshot exhibitionSnapshot) 
+            throws IOException, InterruptedException, FileStorageException {
+        storageEngineDownloads.copyToFolder(exhibitionFolderName + File.separator + RESOURCES_FOLDER_NAME, resourcesPath);
+        List<Space> spaces= spaceRepository.findAllBySpaceStatus(SpaceStatus.PUBLISHED);
+
+        for(Space space : spaces) {
+            renderingManager.createSpaceSnapshot(space, exhibitionFolderName, sequenceHistory);                
+        }
+        SnapshotTask snapshotTask = exhibitionSnapshot.getSnapshotTask();
+        return snapshotTaskRepository.save(snapshotTask);   
+    }
+
+    /**
+     * 
+     * Creates folder for ExhibitionSnapshot. 
+     * 
+     * @param exhibitionSnapshot     {@link ExhibitionSnapshot} for which the folder is created
+     * @param exhibitionFolderName   the folder name of the snapshot
+     * @return the name of the exhibition folder
+     */
+    private String createSnapshotFolder(ExhibitionSnapshot exhibitionSnapshot, String exhibitionFolderName) {
+        exhibitionFolderName = storageEngineDownloads.createFolder(exhibitionFolderName);
+        exhibitionSnapshot.setFolderName(exhibitionFolderName);
+        exhibitionSnapshot = exhibitionSnapshotRepository.save(exhibitionSnapshot);
+        
+        return exhibitionFolderName; 
+    }
+
+    /**
+     * Downloads the given exhibition folder by the given ExhibitionSnapshot id
+     * 
+     * @param id  - exhibition snapshot id
+     * @return - The zip file as a byte array
+     * @throws ExhibitionDownloadNotFoundException
+     * @throws IOException
+     */
+    @Override
+    public byte[] getExhibitionSnapshot(String id) throws ExhibitionSnapshotNotFoundException, IOException {
+        Optional<ExhibitionSnapshot> exhibitionSnapshot = exhibitionSnapshotRepository.findById(id);
+
+        if(!exhibitionSnapshot.isPresent()) {           
+            throw new ExhibitionSnapshotNotFoundException("Exhibition Snapshot not found");             
+        }
+        try {
+            return storageEngineDownloads.getMediaContent("",exhibitionSnapshot.get().getFolderName()+ZIP_FILE_EXTENSION);                
+        }catch(Exception e) {
```

Comment:
is anything actually throwing `Exception`? If so, there should be a comment here. If not, then the specific exception thrown should be caught here.